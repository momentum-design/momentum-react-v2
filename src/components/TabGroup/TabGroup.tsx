import React, { FC, Children, cloneElement, isValidElement, useRef } from 'react';
import classNames from 'classnames';

import { DEFAULTS, STYLE } from './TabGroup.constants';
import { Props } from './TabGroup.types';
import { handleOnKeyDown } from './TabGroup.utils';

import './TabGroup.style.scss';

const TabGroup: FC<Props> = (props: Props) => {
  const {
    ariaLabel,
    ariaLabelledby,
    ariaDescribedby,
    children,
    className,
    id,
    role,
    style,
    orientation,
    spaced,
    ...rest
  } = props;

  const containerRef = useRef(null);

  //Handle onKeyDown events and set default attribute for tab child elements
  const tabsChildren = Children.map(children, (child) => {
    if (isValidElement(child)) {
      return cloneElement(child as React.ReactElement<any>, {
        onKeyDown: (event: KeyboardEvent) => {
          handleOnKeyDown(event, containerRef, orientation);
        },
        role: 'tab',
      });
    };
    return child;
  });

  return (
    <div
      ref={containerRef}
      id={id}
      style={style}
      aria-label={ariaLabel}
      aria-labelledby={ariaLabelledby}
      aria-describedby={ariaDescribedby}
      aria-orientation={orientation}
      className={classNames(STYLE.wrapper, className)}
      data-orientation={orientation || DEFAULTS.ORIENTATION}
      data-spaced={spaced || DEFAULTS.SPACED}
      role={role || DEFAULTS.ROLE}
      {...rest}
    >
      {tabsChildren}
    </div>
  );
};

export default TabGroup;