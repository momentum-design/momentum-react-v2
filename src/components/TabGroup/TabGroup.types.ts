import { CSSProperties, ReactElement } from 'react';

import { TabProps } from '../Tab';

export type SupportedComponents = TabProps;
export type TabGroupOrientation = 'horizontal' | 'vertical';

export interface Props {
  /**
   * aria-label attribute for the tabGrup;
   */
  ariaLabel?: string;

  /**
   * aria-labelledby attribute for the tabGrup;
   */
  ariaLabelledby?: string;

  /**
   * aria-describedby attribute for the tabGrup;
   */
  ariaDescribedby?: string;

  /**
   * Child components of this TabGroup.
   */
  children?: ReactElement<SupportedComponents> | Array<ReactElement<SupportedComponents>>;

  /**
   * Custom class for overriding this component's CSS.
   */
  className?: string;

  /**
   * Custom id for overriding this component's CSS.
   */
  id?: string;

  /**
   * Orientation of TabGroup,  Default: `horizontal`
   * horizontal -> left/right keys to navigate
   * vertical -> up/down keys to navigate
   */
  orientation: TabGroupOrientation;

  /**
   * Role for adding accessibility
   */
  role?: string;

  /**
   * Custom style for overriding this component's CSS.
   */
  style?: CSSProperties;

  /**
   * Whether to include spacing around child components.
   */
  spaced?: boolean;
};
