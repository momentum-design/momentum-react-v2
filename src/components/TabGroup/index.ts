import { default as TabGroup } from './TabGroup';
import * as CONSTANTS from './TabGroup.constants';
import { Props } from './TabGroup.types';

export { CONSTANTS as TAB_GROUP_CONSTANTS };

export type TabGroupProps = Props;

export default TabGroup;