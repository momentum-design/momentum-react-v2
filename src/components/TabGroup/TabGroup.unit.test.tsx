import { TabGroup, TabNext } from "@momentum-ui/react-collaboration"
import { mount } from "enzyme";
import React from "react"
import { DEFAULTS, STYLE } from "./TabGroup.constants";



describe('<TabGroup/>', () => {
    const childrenTemplate = [
        <TabNext key="0">tabA</TabNext>,
        <TabNext key="1">tabB</TabNext>,
        <TabNext key="2">tabC</TabNext>,
    ];

    let container;

    describe('attributes', () => {
        it('should have its wrapper class', () => {
            expect.assertions(1);

            const element = mount(<TabGroup>{childrenTemplate}</TabGroup>)
                .find(TabGroup)
                .getDOMNode();

            expect(element.classList.contains(STYLE.wrapper)).toBe(true);
        });

        it('should have provided class when className is provided', () => {
            expect.assertions(1);

            const className = 'example-class';

            const element = mount(<TabNext className={className} />)
                .find(TabNext)
                .getDOMNode();

            expect(element.classList.contains(className)).toBe(true);
        });

        it('should have provided id when id is provided', () => {
            expect.assertions(1);

            const id = 'example-id';

            const element = mount(<TabNext id={id} />)
                .find(TabNext)
                .getDOMNode();

            expect(element.id).toBe(id);
        });

        it('should have provided style when style is provided', () => {
            expect.assertions(1);

            const style = { color: 'pink' };
            const styleString = 'color: pink;';

            const element = mount(<TabNext style={style} />)
                .find(TabNext)
                .getDOMNode();

            expect(element.getAttribute('style')).toBe(styleString);
        });

        it('should pass spaced prop', () => {
            expect.assertions(1);

            const spaced = !DEFAULTS.SPACED;

            const element = mount(<TabGroup spaced={spaced}>{childrenTemplate}</TabGroup>)
                .find(TabGroup)
                .getDOMNode();

            expect(element.getAttribute('data-spaced')).toBe(`${spaced}`);
        });

        it('should pass role prop', () => {
            expect.assertions(1);

            const role = 'tablist';

            const element = mount(<TabGroup role={role}>{childrenTemplate}</TabGroup>)
                .find(TabGroup)
                .getDOMNode();

            expect(element.getAttribute('role')).toBe(role);
        });

        it('should pass orientation prop', () => {
            expect.assertions(1);

            const orientation = 'vertical';

            const element = mount(<TabGroup orientation={orientation}>{childrenTemplate}</TabGroup>)
                .find(TabGroup)
                .getDOMNode();

            expect(element.getAttribute('data-orientation')).toBe(`${orientation}`);
        });

        it('should pass aria-label prop', () => {
            expect.assertions(1);

            const ariaLabel = 'tabGroup aria label';

            const element = mount(<TabGroup aria-label={ariaLabel}>{childrenTemplate}</TabGroup>)
                .find(TabGroup)
                .getDOMNode();

            expect(element.getAttribute('aria-label')).toBe(`${ariaLabel}`);
        });
    });

    describe('snapshot', () => {
        it('should match snapshot', () => {
            expect.assertions(1);

            container = mount(<TabGroup>{childrenTemplate}</TabGroup>);

            expect(container).toMatchSnapshot();
        });

        it('should match snapshot when spaced', () => {
            expect.assertions(1);

            const spaced = !DEFAULTS.SPACED;

            container = mount(<TabGroup spaced={spaced}>{childrenTemplate}</TabGroup>);

            expect(container).toMatchSnapshot();
        });

        it('should match snapshot with className', () => {
            expect.assertions(1);

            const className = 'example-class';

            const container = mount(<TabNext className={className} />);

            expect(container).toMatchSnapshot();
        });

        it('should match snapshot with id', () => {
            expect.assertions(1);

            const id = 'example-id';

            const container = mount(<TabNext id={id} />);

            expect(container).toMatchSnapshot();
        });

        it('should match snapshot with style', () => {
            expect.assertions(1);

            const style = { color: 'pink' };

            const container = mount(<TabNext style={style} />);

            expect(container).toMatchSnapshot();
        });

        it('should match snapshot with role', () => {
            expect.assertions(1);

            const role = 'tablist';

            container = mount(<TabGroup role={role}>{childrenTemplate}</TabGroup>);

            expect(container).toMatchSnapshot();
        });

        it('should match snapshot with orientation = vertical', () => {
            expect.assertions(1);

            const orientation = 'vertical';

            container = mount(<TabGroup orientation={orientation}>{childrenTemplate}</TabGroup>);

            expect(container).toMatchSnapshot();
        });

        it('should match snapshot with aria-label', () => {
            expect.assertions(1);

            const ariaLabel = 'my aria label';

            container = mount(<TabGroup aria-label={ariaLabel}>{childrenTemplate}</TabGroup>);

            expect(container).toMatchSnapshot();
        });
    });
})