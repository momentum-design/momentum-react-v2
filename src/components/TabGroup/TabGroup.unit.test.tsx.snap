// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabGroup/> snapshot should match snapshot 1`] = `
<TabGroup
  orientation="horizontal"
>
  <div
    aria-orientation="horizontal"
    className="md-tab-group-wrapper"
    data-orientation="horizontal"
    data-spaced="false"
    role="tablist"
  >
    <Tab
      key=".$0"
      onKeyDown={[Function]}
      role="tab"
    >
      <ButtonSimple
        className="md-tab-wrapper"
        data-active={false}
        data-disabled={false}
        onKeyDown={[Function]}
        role="tab"
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              type="button"
            >
              <PrimitiveConverter
                fontStyle="subheader-secondary"
              >
                <Text
                  type="subheader-secondary"
                >
                  <h3
                    className="md-text-wrapper"
                    data-type="subheader-secondary"
                  >
                    tabA
                  </h3>
                </Text>
              </PrimitiveConverter>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </Tab>
    <Tab
      key=".$1"
      onKeyDown={[Function]}
      role="tab"
    >
      <ButtonSimple
        className="md-tab-wrapper"
        data-active={false}
        data-disabled={false}
        onKeyDown={[Function]}
        role="tab"
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              type="button"
            >
              <PrimitiveConverter
                fontStyle="subheader-secondary"
              >
                <Text
                  type="subheader-secondary"
                >
                  <h3
                    className="md-text-wrapper"
                    data-type="subheader-secondary"
                  >
                    tabB
                  </h3>
                </Text>
              </PrimitiveConverter>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </Tab>
    <Tab
      key=".$2"
      onKeyDown={[Function]}
      role="tab"
    >
      <ButtonSimple
        className="md-tab-wrapper"
        data-active={false}
        data-disabled={false}
        onKeyDown={[Function]}
        role="tab"
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              type="button"
            >
              <PrimitiveConverter
                fontStyle="subheader-secondary"
              >
                <Text
                  type="subheader-secondary"
                >
                  <h3
                    className="md-text-wrapper"
                    data-type="subheader-secondary"
                  >
                    tabC
                  </h3>
                </Text>
              </PrimitiveConverter>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </Tab>
  </div>
</TabGroup>
`;

exports[`<TabGroup/> snapshot should match snapshot when spaced 1`] = `
<TabGroup
  orientation="horizontal"
  spaced={false}
>
  <div
    aria-orientation="horizontal"
    className="md-tab-group-wrapper"
    data-orientation="horizontal"
    data-spaced="false"
    role="tablist"
  >
    <Tab
      key=".$0"
      onKeyDown={[Function]}
      role="tab"
    >
      <ButtonSimple
        className="md-tab-wrapper"
        data-active={false}
        data-disabled={false}
        onKeyDown={[Function]}
        role="tab"
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              type="button"
            >
              <PrimitiveConverter
                fontStyle="subheader-secondary"
              >
                <Text
                  type="subheader-secondary"
                >
                  <h3
                    className="md-text-wrapper"
                    data-type="subheader-secondary"
                  >
                    tabA
                  </h3>
                </Text>
              </PrimitiveConverter>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </Tab>
    <Tab
      key=".$1"
      onKeyDown={[Function]}
      role="tab"
    >
      <ButtonSimple
        className="md-tab-wrapper"
        data-active={false}
        data-disabled={false}
        onKeyDown={[Function]}
        role="tab"
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              type="button"
            >
              <PrimitiveConverter
                fontStyle="subheader-secondary"
              >
                <Text
                  type="subheader-secondary"
                >
                  <h3
                    className="md-text-wrapper"
                    data-type="subheader-secondary"
                  >
                    tabB
                  </h3>
                </Text>
              </PrimitiveConverter>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </Tab>
    <Tab
      key=".$2"
      onKeyDown={[Function]}
      role="tab"
    >
      <ButtonSimple
        className="md-tab-wrapper"
        data-active={false}
        data-disabled={false}
        onKeyDown={[Function]}
        role="tab"
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              type="button"
            >
              <PrimitiveConverter
                fontStyle="subheader-secondary"
              >
                <Text
                  type="subheader-secondary"
                >
                  <h3
                    className="md-text-wrapper"
                    data-type="subheader-secondary"
                  >
                    tabC
                  </h3>
                </Text>
              </PrimitiveConverter>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </Tab>
  </div>
</TabGroup>
`;

exports[`<TabGroup/> snapshot should match snapshot with aria-label 1`] = `
<TabGroup
  aria-label="my aria label"
  orientation="horizontal"
>
  <div
    aria-label="my aria label"
    aria-orientation="horizontal"
    className="md-tab-group-wrapper"
    data-orientation="horizontal"
    data-spaced="false"
    role="tablist"
  >
    <Tab
      key=".$0"
      onKeyDown={[Function]}
      role="tab"
    >
      <ButtonSimple
        className="md-tab-wrapper"
        data-active={false}
        data-disabled={false}
        onKeyDown={[Function]}
        role="tab"
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              type="button"
            >
              <PrimitiveConverter
                fontStyle="subheader-secondary"
              >
                <Text
                  type="subheader-secondary"
                >
                  <h3
                    className="md-text-wrapper"
                    data-type="subheader-secondary"
                  >
                    tabA
                  </h3>
                </Text>
              </PrimitiveConverter>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </Tab>
    <Tab
      key=".$1"
      onKeyDown={[Function]}
      role="tab"
    >
      <ButtonSimple
        className="md-tab-wrapper"
        data-active={false}
        data-disabled={false}
        onKeyDown={[Function]}
        role="tab"
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              type="button"
            >
              <PrimitiveConverter
                fontStyle="subheader-secondary"
              >
                <Text
                  type="subheader-secondary"
                >
                  <h3
                    className="md-text-wrapper"
                    data-type="subheader-secondary"
                  >
                    tabB
                  </h3>
                </Text>
              </PrimitiveConverter>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </Tab>
    <Tab
      key=".$2"
      onKeyDown={[Function]}
      role="tab"
    >
      <ButtonSimple
        className="md-tab-wrapper"
        data-active={false}
        data-disabled={false}
        onKeyDown={[Function]}
        role="tab"
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              type="button"
            >
              <PrimitiveConverter
                fontStyle="subheader-secondary"
              >
                <Text
                  type="subheader-secondary"
                >
                  <h3
                    className="md-text-wrapper"
                    data-type="subheader-secondary"
                  >
                    tabC
                  </h3>
                </Text>
              </PrimitiveConverter>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </Tab>
  </div>
</TabGroup>
`;

exports[`<TabGroup/> snapshot should match snapshot with className 1`] = `
<Tab
  className="example-class"
>
  <ButtonSimple
    className="md-tab-wrapper example-class"
    data-active={false}
    data-disabled={false}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <button
          className="md-tab-wrapper example-class md-button-simple-wrapper"
          data-active={false}
          data-disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          type="button"
        />
      </FocusRing>
    </FocusRing>
  </ButtonSimple>
</Tab>
`;

exports[`<TabGroup/> snapshot should match snapshot with id 1`] = `
<Tab
  id="example-id"
>
  <ButtonSimple
    className="md-tab-wrapper"
    data-active={false}
    data-disabled={false}
    id="example-id"
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <button
          className="md-tab-wrapper md-button-simple-wrapper"
          data-active={false}
          data-disabled={false}
          id="example-id"
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          type="button"
        />
      </FocusRing>
    </FocusRing>
  </ButtonSimple>
</Tab>
`;

exports[`<TabGroup/> snapshot should match snapshot with orientation = vertical 1`] = `
<TabGroup
  orientation="vertical"
>
  <div
    aria-orientation="vertical"
    className="md-tab-group-wrapper"
    data-orientation="vertical"
    data-spaced="false"
    role="tablist"
  >
    <Tab
      key=".$0"
      onKeyDown={[Function]}
      role="tab"
    >
      <ButtonSimple
        className="md-tab-wrapper"
        data-active={false}
        data-disabled={false}
        onKeyDown={[Function]}
        role="tab"
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              type="button"
            >
              <PrimitiveConverter
                fontStyle="subheader-secondary"
              >
                <Text
                  type="subheader-secondary"
                >
                  <h3
                    className="md-text-wrapper"
                    data-type="subheader-secondary"
                  >
                    tabA
                  </h3>
                </Text>
              </PrimitiveConverter>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </Tab>
    <Tab
      key=".$1"
      onKeyDown={[Function]}
      role="tab"
    >
      <ButtonSimple
        className="md-tab-wrapper"
        data-active={false}
        data-disabled={false}
        onKeyDown={[Function]}
        role="tab"
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              type="button"
            >
              <PrimitiveConverter
                fontStyle="subheader-secondary"
              >
                <Text
                  type="subheader-secondary"
                >
                  <h3
                    className="md-text-wrapper"
                    data-type="subheader-secondary"
                  >
                    tabB
                  </h3>
                </Text>
              </PrimitiveConverter>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </Tab>
    <Tab
      key=".$2"
      onKeyDown={[Function]}
      role="tab"
    >
      <ButtonSimple
        className="md-tab-wrapper"
        data-active={false}
        data-disabled={false}
        onKeyDown={[Function]}
        role="tab"
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              type="button"
            >
              <PrimitiveConverter
                fontStyle="subheader-secondary"
              >
                <Text
                  type="subheader-secondary"
                >
                  <h3
                    className="md-text-wrapper"
                    data-type="subheader-secondary"
                  >
                    tabC
                  </h3>
                </Text>
              </PrimitiveConverter>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </Tab>
  </div>
</TabGroup>
`;

exports[`<TabGroup/> snapshot should match snapshot with role 1`] = `
<TabGroup
  orientation="horizontal"
  role="tablist"
>
  <div
    aria-orientation="horizontal"
    className="md-tab-group-wrapper"
    data-orientation="horizontal"
    data-spaced="false"
    role="tablist"
  >
    <Tab
      key=".$0"
      onKeyDown={[Function]}
      role="tab"
    >
      <ButtonSimple
        className="md-tab-wrapper"
        data-active={false}
        data-disabled={false}
        onKeyDown={[Function]}
        role="tab"
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              type="button"
            >
              <PrimitiveConverter
                fontStyle="subheader-secondary"
              >
                <Text
                  type="subheader-secondary"
                >
                  <h3
                    className="md-text-wrapper"
                    data-type="subheader-secondary"
                  >
                    tabA
                  </h3>
                </Text>
              </PrimitiveConverter>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </Tab>
    <Tab
      key=".$1"
      onKeyDown={[Function]}
      role="tab"
    >
      <ButtonSimple
        className="md-tab-wrapper"
        data-active={false}
        data-disabled={false}
        onKeyDown={[Function]}
        role="tab"
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              type="button"
            >
              <PrimitiveConverter
                fontStyle="subheader-secondary"
              >
                <Text
                  type="subheader-secondary"
                >
                  <h3
                    className="md-text-wrapper"
                    data-type="subheader-secondary"
                  >
                    tabB
                  </h3>
                </Text>
              </PrimitiveConverter>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </Tab>
    <Tab
      key=".$2"
      onKeyDown={[Function]}
      role="tab"
    >
      <ButtonSimple
        className="md-tab-wrapper"
        data-active={false}
        data-disabled={false}
        onKeyDown={[Function]}
        role="tab"
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              type="button"
            >
              <PrimitiveConverter
                fontStyle="subheader-secondary"
              >
                <Text
                  type="subheader-secondary"
                >
                  <h3
                    className="md-text-wrapper"
                    data-type="subheader-secondary"
                  >
                    tabC
                  </h3>
                </Text>
              </PrimitiveConverter>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </Tab>
  </div>
</TabGroup>
`;

exports[`<TabGroup/> snapshot should match snapshot with style 1`] = `
<Tab
  style={
    Object {
      "color": "pink",
    }
  }
>
  <ButtonSimple
    className="md-tab-wrapper"
    data-active={false}
    data-disabled={false}
    style={
      Object {
        "color": "pink",
      }
    }
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <button
          className="md-tab-wrapper md-button-simple-wrapper"
          data-active={false}
          data-disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "color": "pink",
            }
          }
          type="button"
        />
      </FocusRing>
    </FocusRing>
  </ButtonSimple>
</Tab>
`;
